tools/make/golang.mk

```diff
@@ -32,7 +32,7 @@ go.build.%:
 
 # Build the envoy-gateway binaries in the hosted platforms.
 .PHONY: go.build
-go.build: $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
+go.build: go.generate $(addprefix go.build., $(addprefix $(PLATFORM)., $(BINS)))
```

Comment:
this shouldnt be dependancy imo

Reply:
you should run `go generate` before `go build`

Reply:
we already generate a lot of go code today 
https://github.com/envoyproxy/gateway/blob/630cae98c9fb6e66f303ef9174876aa6d21186b6/tools/make/kube.mk#L32
I suggest we move this target into `common.mk` and also add a `go generate` step in there

Reply:
yeah, that make sense

Reply:
updated, PTAL

Reply:
thanks, it looks good, can you confirm that `make geo-check` works with no hiccups ?
can you also update the print statement in https://github.com/envoyproxy/gateway/blob/630cae98c9fb6e66f303ef9174876aa6d21186b6/tools/make/lint.mk#L72 ?

Reply:
the output looks expected to me
```
make[1]: Entering directory '/root/go/src/github.com/envoyproxy/gateway'
===========> Running kube-generate ... 
tools/bin/controller-gen object:headerFile="/root/go/src/github.com/envoyproxy/gateway/tools/boilerplate/boilerplate.generatego.txt",year=2023 paths="{/root/go/src/github.com/envoyproxy/gateway/api/...,/root/go/src/github.com/envoyproxy/gateway/internal/ir/...,/root/go/src/github.com/envoyproxy/gateway/internal/gatewayapi/...}"
===========> Running go.generate ... 
go generate ./...
===========> Running manifests ... 
tools/bin/controller-gen rbac:roleName=envoy-gateway-role crd webhook paths="./..." output:crd:artifacts:config=charts/eg/crds/generated output:rbac:artifacts:config=charts/eg/templates/generated/rbac output:webhook:artifacts:config=charts/eg/templates/generated/webhook
===========> Running gen-check ... 
make[1]: Leaving directory '/root/go/src/github.com/envoyproxy/gateway'
```